docs/source/python/dlpack.rst

```diff
@@ -0,0 +1,82 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _pyarrow-dlpack:
+
+The DLPack Protocol
+===================
+
+Producing side of the DLPack Protocol is implemented for ``pa.Array``
+and can be used to interchange data between PyArrow and other tensor
+libraries. The data structures that are supported in the implementation
+of the protocol are integer, unsigned integer and float arrays. The
+protocol has no missing data support meaning PyArrow arrays with
+validity mask can not be used to transfer data through the DLPack
+protocol. Currently Arrow implementation of the protocol only supports
+data on a CPU device.
+
+The DLPack Protocol is
```

Comment:
Some text suggestions: I would maybe first explain DLPack before saying what is implemented by pyarrow (so moving this paragraph first), and then within this paragraph I think you can also explain first what DLPack is and then mention it is selected by the consortium.